### ISSUE TITLE
Is pivot_longer(names_to = NULL) an official spec?

### ISSUE THREAD

**Issue by yutannihilation** [tidyverse/tidyr#1120]
While #962 says so, I found it's documented nowhere; NEWS describes it's `character()`, not `NULL`,

> `pivot_longer()` now supports `names_to = character()` which prevents the name column from being created (#961).

and the document of `pivot_longer()` says nothing, while the current behavior seems to support both. I feel `NULL` is more appropriate as a user-facing interface. May I expect this is an official spec of `names_to` and is it okay to send a documentation pull request about this?

``` r
library(tidyr)

df <- tibble(x = 1)
build_longer_spec(df, x, names_to = character())
#> # A tibble: 1 x 2
#>   .name .value
#>   <chr> <chr> 
#> 1 x     value
build_longer_spec(df, x, names_to = NULL)
#> # A tibble: 1 x 2
#>   .name .value
#>   <chr> <chr> 
#> 1 x     value
```

<sup>Created on 2021-05-15 by the [reprex package](https://reprex.tidyverse.org) (v2.0.0)</sup>

**Comment by hadley** [comment]
`character()` definitely feels more correct, but I agree that `NULL` is friendlier for a using facing functions. Do you have any thoughts either way @DavisVaughan?

**Comment by DavisVaughan** [comment]
I also considered if it should really be `zap()` to zap away the new column (somewhat like `vec_rbind(.names_to = zap())`), but I think that is too low level.

I generally like reserving `NULL` to only be used as the default argument, rather than as something with special meaning. In the past we talked about this being identical to "the user didn't supply this argument at all".

So I'd lean towards `character()` with a refresh of the `names_to` documentation, clearly stating what happens in the length `0`, `1`, and `>1` character vector cases

**Comment by yutannihilation** [comment]
> I generally like reserving `NULL` to only be used as the default argument, rather than as something with special meaning. In the past we talked about this being identical to "the user didn't supply this argument at all".

I agree with you to some extent, but I'd argue there are many cases when explicit `NULL` makes sense (e.g. `setNames(x, NULL)`) to express the intent to remove something. (Not yet sure if it's the case this time, though...)
